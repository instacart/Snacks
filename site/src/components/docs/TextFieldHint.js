import React, {Fragment} from 'react'
import Playground from '../Playground'

export default () => (

TextFieldHint is part of how TextField works. See TextField to see how TextFieldHint is used.

)
