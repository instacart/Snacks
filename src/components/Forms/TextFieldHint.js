import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { colors } from '../../styles'

const styles = {
  root: {
    color: `${colors.GRAY_74}`,
    cursor: 'inherit',
    fontSize: '16px',
    position: 'absolute',
    opacity: 0,
    transform: 'scale(1) translate(9px, 26px)',
    transformOrigin: 'left top',
    transition: 'visibility 0ms linear 250ms, opacity 250ms cubic-bezier(0.23, 1, 0.32, 1) 0ms',
    pointerEvents: 'none',
    zIndex: 1,
    visibility: 'hidden',
  },
  show: {
    opacity: 1,
    transition: 'visibility 0ms linear 0ms, opacity 250ms cubic-bezier(0.23, 1, 0.32, 1) 0ms',
    visibility: 'visible',
  },
  disabled: {
    cursor: 'not-allowed',
  },
}

// eslint-disable-next-line react/prefer-stateless-function
class TextFieldHint extends Component {
  static propTypes = {
    className: PropTypes.string,
    /** Hint Text */
    text: PropTypes.string.isRequired,
    /** Disabled styling */
    disabled: PropTypes.bool,
    /** Show the hint text */
    show: PropTypes.bool,
    /**
     * Optional style overrides merged into emotion css
     *
     * @deprecated
     * This prop exists for backwards compatibility and will be
     * removed in a future version
     */
    style: PropTypes.object,
    /** A uniq id */
    inputId: PropTypes.string,
  }

  static defaultProps = {
    disabled: false,
  }

  render() {
    const { className, disabled, show, style, text, inputId } = this.props

    return (
      <div
        className={className}
        id={inputId}
        css={[styles.root, style, disabled && styles.disabled, show && styles.show]}
      >
        {text}
      </div>
    )
  }
}

export default TextFieldHint
