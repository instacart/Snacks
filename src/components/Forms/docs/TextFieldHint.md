TextFieldHint is part of how TextField works. See TextField to see how TextFieldHint is used.
