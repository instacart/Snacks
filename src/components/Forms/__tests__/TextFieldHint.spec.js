import React       from 'react'
import renderer    from 'react-test-renderer'
import { StyleRoot } from 'radium'
import TextFieldHint from '../TextFieldHint'

it('renders TextFieldHint correctly', () => {
  const tree = renderer.create(
    <StyleRoot>
      <div>
        <TextFieldHint text="Text field hint text" show />
      </div>
    </StyleRoot>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})



it('renders TextFieldHint when show is false', () => {
  const tree = renderer.create(
    <StyleRoot>
      <div>
        <TextFieldHint text="Text field hint text" show={false} />
      </div>
    </StyleRoot>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders TextFieldHint when disabled', () => {
  const tree = renderer.create(
    <StyleRoot>
      <div>
        <TextFieldHint text="Text field hint text" show disabled />
      </div>
    </StyleRoot>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
