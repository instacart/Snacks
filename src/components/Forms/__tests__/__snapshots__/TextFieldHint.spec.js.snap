// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TextFieldHint correctly 1`] = `
<div
  data-radium={true}
>
  <div>
    <div
      data-radium={true}
      style={
        Object {
          "color": "#BDBDBD",
          "cursor": "inherit",
          "fontSize": "16px",
          "opacity": 1,
          "pointerEvents": "none",
          "position": "absolute",
          "transform": "scale(1) translate(9px, 23px)",
          "transformOrigin": "left top",
          "transition": "opacity 250ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "zIndex": 1,
        }
      }
    >
      Text field hint text
    </div>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`renders TextFieldHint when disabled 1`] = `
<div
  data-radium={true}
>
  <div>
    <div
      data-radium={true}
      style={
        Object {
          "color": "#BDBDBD",
          "cursor": "not-allowed",
          "fontSize": "16px",
          "opacity": 1,
          "pointerEvents": "none",
          "position": "absolute",
          "transform": "scale(1) translate(9px, 23px)",
          "transformOrigin": "left top",
          "transition": "opacity 250ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "zIndex": 1,
        }
      }
    >
      Text field hint text
    </div>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`renders TextFieldHint when show is false 1`] = `
<div
  data-radium={true}
>
  <div>
    <div
      data-radium={true}
      style={
        Object {
          "color": "#BDBDBD",
          "cursor": "inherit",
          "fontSize": "16px",
          "opacity": 0,
          "pointerEvents": "none",
          "position": "absolute",
          "transform": "scale(1) translate(9px, 23px)",
          "transformOrigin": "left top",
          "transition": "opacity 250ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "zIndex": 1,
        }
      }
    >
      Text field hint text
    </div>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;
